File: simexplorer-ide/simexplorer-ide-ant/org.openmole.ui/src/org/openmole/ui/workflow/implementation/paint/MyWidget.java
/*
 *  Copyright (C) 2010 leclaire
 *
 *  This program is free software: you can redistribute it and/or modify
 *  it under the terms of the GNU General Public License as published by
 *  the Free Software Foundation, either version 3 of the License, or
 *  (at your option) any later version.
 *
 *  This program is distributed in the hope that it will be useful,
 *  but WITHOUT ANY WARRANTY; without even the implied warranty of
 *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 *  GNU General Public License for more details.
 *
 *  You should have received a copy of the GNU General Public License
 *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */
package org.openmole.ui.workflow.implementation.paint;

import java.awt.Color;
import java.awt.Container;
import java.awt.Cursor;
import org.netbeans.api.visual.widget.*;
import java.awt.Graphics2D;
import java.awt.Image;
import java.awt.Rectangle;
import org.openmole.ui.commons.ApplicationCustomize;
import org.openmole.ui.workflow.implementation.MoleScene;

/**
 *
 * @author Mathieu Leclaire <mathieu.leclaire@openmole.fr>
 */
public class MyWidget extends Widget {

    private Color backgroundCol;
    protected  Color borderCol;
    private Image backgroundImaqe;
    protected Rectangle bodyArea = new Rectangle();
    private Rectangle widgetArea = new Rectangle();
    private Rectangle titleArea;
    private boolean title = false;
    private boolean image = false;
    private String titleString;
    private Container dataTableContainer = new Container();
    MoleScene scene;

    public MyWidget(MoleScene scene,
            Color col) {
        super(scene);
        this.backgroundCol = col;
        Rectangle bodyrect = new Rectangle(0, 0,
                ApplicationCustomize.TASK_CONTAINER_WIDTH,
                ApplicationCustomize.TASK_CONTAINER_HEIGHT);

        Rectangle widgetrect = new Rectangle(-8, -1,
                ApplicationCustomize.TASK_CONTAINER_WIDTH + 16,
                ApplicationCustomize.TASK_CONTAINER_HEIGHT + 2);
        bodyArea.setBounds(bodyrect);
        widgetArea.setBounds(widgetrect);

        dataTableContainer.setCursor(Cursor.getPredefinedCursor(Cursor.HAND_CURSOR));
        setPreferredBounds(widgetArea);
    }

    public void setBackgroundCol(Color backgroundCol) {
        this.backgroundCol = backgroundCol;
    }

    public void setBorderCol(Color borderCol) {
        this.borderCol = borderCol;
    }

    public void setBackgroundImaqe(Image backgroundImaqe) {
        this.backgroundImaqe = backgroundImaqe;
    }

    public MyWidget(MoleScene scene,
            Color backgroundCol,
            Color borderCol) {
        this(scene, backgroundCol);
        this.borderCol = borderCol;
    }

    public MyWidget(MoleScene scene,
            Color backgroundCol,
            Color borderCol,
            Image backgroundImaqe) {
        this(scene, backgroundCol, borderCol);
        this.image = true;
        this.backgroundImaqe = backgroundImaqe;
    }

    private void enlargeWidgetArea(Rectangle addedArea) {
        widgetArea.height += addedArea.height;
        widgetArea.y -= addedArea.y;
    }

    public void setTitle(String title){
        this.titleString = title;
    }

    public void addTitle(String titleString) {
        titleArea = new Rectangle(0, 0,
                ApplicationCustomize.TASK_CONTAINER_WIDTH,
                ApplicationCustomize.TASK_TITLE_HEIGHT);

        this.title = true;
        this.titleString = titleString;

        enlargeWidgetArea(titleArea);
        setPreferredBounds(widgetArea);
    }

    @Override
    protected void paintWidget() {
        Graphics2D graphics = getGraphics();
        graphics.setColor(backgroundCol);
        graphics.fill(bodyArea);

        graphics.setColor(borderCol);

        if (title) {
            graphics.fill(titleArea);
            graphics.setColor(Color.WHITE);
            graphics.drawString(titleString, 10, 15);
        }

        if (image){
        graphics.drawImage(backgroundImaqe,
                           ApplicationCustomize.TASK_IMAGE_WIDTH_OFFSET,
                           ApplicationCustomize.TASK_IMAGE_HEIGHT_OFFSET,
                           ApplicationCustomize.TASK_IMAGE_WIDTH,
                           ApplicationCustomize.TASK_IMAGE_HEIGHT,
                           backgroundCol,
                           new Container());
        }
    }
}

File: platform/core/org.openmole.ide.core.implementation/src/main/scala/org/openmole/ide/core/implementation/workflow/MyWidget.scala
/*
 * Copyright (C) 2011 Mathieu leclaire <mathieu.leclaire at openmole.org>
 *
 * This program is free software: you can redistribute it and/or modify
 * it under the terms of the GNU Affero General Public License as published by
 * the Free Software Foundation, either version 3 of the License, or
 * (at your option) any later version.
 *
 * This program is distributed in the hope that it will be useful,
 * but WITHOUT ANY WARRANTY; without even the implied warranty of
 * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
 * GNU Affero General Public License for more details.
 *
 * You should have received a copy of the GNU General Public License
 * along with this program.  If not, see <http://www.gnu.org/licenses/>.
 */

package org.openmole.ide.core.implementation.workflow

import java.awt.Color
import java.awt.Rectangle
import java.awt.Font
import java.awt.Graphics2D
import org.netbeans.api.visual.widget._
import org.openmole.ide.core.model.commons.Constants._
import org.openmole.ide.core.model.workflow.ICapsuleUI
import java.awt.BasicStroke
import org.openmole.ide.core.model.workflow.IMoleScene
import org.openmole.ide.core.model.panel.PanelMode._

class MyWidget(scene: IMoleScene,capsule: ICapsuleUI) extends Widget(scene.graphScene) {

  var taskHeight= TASK_CONTAINER_HEIGHT
  val bodyArea = new Rectangle
  val titleArea = new Rectangle
  
  override def paintWidget= {
    val graphics= getGraphics.asInstanceOf[Graphics2D]
    if(capsule.dataProxy.isDefined){
      val tpud = capsule.dataProxy.get.dataUI
      drawBox(graphics,tpud.backgroundColor,tpud.borderColor)
      graphics.setColor(Color.WHITE)
      graphics.setFont(new Font("Ubuntu", Font.PLAIN, 15))
      graphics.drawString(tpud.name, 10, 15)
    }
    else drawBox(graphics,new Color(204,204,204,128),new Color(204,204,204))
  }
  
  def drawBox(graphics: Graphics2D,c1: Color, c2: Color) = {
    scene match {
      case x: BuildMoleScene=> graphics.setColor(c1)
      case _=> graphics.setColor(new Color(215,238,244,64))
    }
    graphics.fill(bodyArea)
    scene match {
      case x: BuildMoleScene=> graphics.setColor(c2)
      case _=> graphics.setColor(new Color(44,137,160,64))
    }
    graphics.draw(new BasicStroke(1.3f, 1, 1).createStrokedShape(bodyArea))
  }
  
}
